wallets/react/src/legacy/useLegacyProviders.ts

```diff
@@ -30,22 +42,48 @@ export function useLegacyProviders(props: ProviderProps): ProviderContext {
   const wallets = checkWalletProviders(listOfProviders);
 
   useAutoConnect({
-    wallets,
     allBlockChains: props.allBlockChains,
     autoConnect: props.autoConnect,
-    getWalletInstanceFromLegacy: getWalletInstance,
+    autoConnectHandler: async () => autoConnect(wallets, getWalletInstance),
   });
 
   // Final API we put in context and it will be available to use for users.
-  // eslint-disable-next-line react/jsx-no-constructed-context-values
   const api: ProviderContext = {
-    async connect(type, network, namespaces) {
+    async connect(type, namespaces) {
       const wallet = wallets.get(type);
       if (!wallet) {
         throw new Error(`You should add ${type} to provider first.`);
       }
+
+      /**
+       * Discover mode has a meaning in hub, so we are considering whenever a namespace with DISCOVER_MODE reaches here,
+       * we can ignore it and don't pass it to provider.
+       */
+      const namespacesForConnect = namespaces?.filter(
+        (
+          ns
+        ): ns is Exclude<
+          LegacyNamespaceInputForConnect,
+          LegacyNamespaceInputWithDiscoverMode
+        > => {
+          return ns.namespace !== 'DISCOVER_MODE';
+        }
+      );
+      // Legacy providers doesn't implemented multiple namespaces, so it will always be one value.
```

Comment:
I don't think that this a valid assumption, why should there be such limitation for legacy providers to only implement one single namespace?